src/Configuration/Config.AzureKeyVault/src/AzureKeyVaultConfigurationProvider.cs

```diff
@@ -78,6 +108,43 @@ private async Task LoadAsync()
             } while (secrets != null);
 
             Data = data;
+            OnReload();
+
+            if (_reloadOnChange && _pollingTask == null)
+            {
+                _pollingTask = PollForSecretChangesAsync(_reloadPollDelay, _cancellationToken.Token);
+            }
+        }
+
+        private async Task<bool> ShouldReloadAsync()
+        {
+            var secrets = await _client.GetSecretsAsync(_vault).ConfigureAwait(false);
```

Comment:
This doesn't seem to account for pagination. GetSecretsAsync returns a single page of secrets you have to call GetSecretsNextAsync to get more.